app/services/hide_from_search_validator.rb

```diff
@@ -0,0 +1,20 @@
+class HideFromSearchValidator
+  attr_reader :base_path
+
+  def initialize(base_path)
+    @base_path = base_path
+  end
+
+  def valid?(document)
```

Comment:
(I personally find this naming a bit obscure: `valid?` usually refers to if an object is syntactically correct, rather than if it should or shouldn't show something. I'd personally prefer something like `should_show?`, but fine to leave)